### ISSUE TITLE
Monitoring mqttwarn itself

### ISSUE THREAD

**Issue by rgitzel** [jpmens/mqttwarn#328]
I’ve about a dozen Sonoff devices (running the wonderful [Tasmota](https://github.com/arendst/Sonoff-Tasmota)) scattered around my house and yard, now, and mqttwarn processes data from all of them, mostly to transform Tasmota's JSON messages for sensor readings and wifi signal strengths into InfluxDb entries (which are published back to MQTT and picked up by Telegraf).

One concern I’ve had is detecting when any of this ever-increasing set of devices is having problems (for instance, the TH16 controlling our attic AC conked out on the weekend, and I only noticed by fluke).

It occurred to me that since mqttwarn is already central to this mesh of devices, I could use it to keep tabs on it all. Thing is, how?

One approach is to collect real-time statistics… so I’ve added the ability for mqttwarn to push some into InfluxDb: https://github.com/rgitzel/mqttwarn/pull/1/files

WARNING:  I am *not* a Python dev, obviously.  (But I do do this for a living in other languages.)  And the branch is not yet viable to be added.  But it’s a start for the adventurous.

Using that code, I’ve been able to make this dashboard in Grafana:

<img width="1057" alt="4fa4cdbb-1562-45d2-aeec-6bcd0c2983f0" src="https://user-images.githubusercontent.com/688749/45167072-403d0700-b1ad-11e8-9939-ab8ce42130e8.png">

All of the readings are tagged with the full topic string, with each ‘level’ in the topic (e.g. “a/b/c” would be tagged with ‘level0=a’, ‘level1=b’ and ‘level2=c”), and with the service, if applicable.

Thus a couple of the graphs in that dashboard break down messages by their Tasmota “type” (topics are typically something like “tele/{device}/SENSOR”).  Since the particular device name is 'level1' a graph could should values for all the devices on one graph, which would be crazy; but you _could_ tailor a graph's query to include only certain devices by name.

What do these graphs mean?  I’m really not sure, yet.  I can’t even say they are completely accurate.  But some oddities look worth investigating:
* I would expect numbers of different types of messages to be consistent if the number of devices doesn’t change.  So what’s with the ‘STATUS11’ messages?  Why only certain times of day?
* What happened around 19:00 to slow processing?  I’m running *all* of this on my Synology NAS in Docker containers… maybe the NAS got busy around then?  

Anyway, I’ll keep tinkering. I’d love any feedback (more about what it’s doing than about code details, I’ll get to that later), particularly suggestions for things to collect.  Errors are definitely one thing I'd like to add, probably broken down into a few categories like app failure, bad messages, inability to notify a service, etc.

**Comment by jpmens** [comment]
@sumnerboy12 Care to chime in to the above?

**Comment by sumnerboy12** [comment]
I went thru this process a few years ago - realising I had too many devices to monitor _by-eye_ so thanks to a suggestion by @jpmens I went ahead and installed `icinga2`. I then added an `icinga2` service to `mqttwarn` and now I can quite easily setup host/service checks in `icinga2` based on MQTT publishes received by `mqttwarn`. 

I setup passive checks in `icinga2` which are updated by a publish to a particular MQTT topic. If a pub isn't received in a configurable period of time, `icinga2` starts raising warns and crits which I can configure to notify me via Slack or Pushover.
